src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift

```diff
@@ -0,0 +1,41 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+ protocol QRCodeErrorCorrectionLevelProviding {
+	func errorCorrectionLevel(
+		appConfigurationProvider: AppConfigurationProviding,
+		onCompletion: @escaping((String) -> Void)
+	)
+ }
+
+class QRCodeErrorCorrectionLevelProvider {
+	private var subscriptions: Set<AnyCancellable> = []
+
+	func errorCorrectionLevel(
+		appConfigurationProvider: AppConfigurationProviding,
+		onCompletion: @escaping((String) -> Void)
+	) {
+		appConfigurationProvider.appConfiguration().sink { appConfig in
+			let qrCodeErrorCorrectionLevel = appConfig.presenceTracingParameters.qrCodeErrorCorrectionLevel
+			var mappedErrorCorrectionLevel = "M"
+			
```

Comment:
this is the same regarding the enum maybe here is a good place to do so.
`var mappedErrorCorrectionLevel: QRCodeErrorCorrectionLevel = .medium`

Reply:
It doesn’t make sense, we need letters and not numbers for QR code generation.

Reply:
you can declare enums with string property and use the cases as string or at least their raw values

Reply:
or you could pass it as associated value in the enum case